usr/src/uts/common/fs/zfs/zil.c

```diff
@@ -908,9 +918,13 @@ zil_lwb_write_init(zilog_t *zilog, lwb_t *lwb)
 		    ZIO_FLAG_CANFAIL);
 	}
 	if (lwb->lwb_zio == NULL) {
+		if (zilog->zl_cur_used <= zil_slog_bulk || !lwb->lwb_slog)
```

Comment:
I would put the `lwb_slog` check first, so that it is more clear that we do not care about the used size when we are not using slog.